tools/kicad_rs/src/eval/index.rs

```diff
@@ -0,0 +1,92 @@
+use crate::eval::entry::Entry;
+use crate::eval::path::Path;
+use evalexpr::{Context, ContextWithMutableVariables, EvalexprError, EvalexprResult, Value};
+use std::collections::HashMap;
+
+// The default attribute is signified by an empty string
+const DEFAULT_ATTR: String = String::new();
+
+pub type ComponentIndex<'a> = HashMap<String, Entry<'a>>;
+
+#[derive(Default, Debug)]
+pub struct SheetIndex<'a> {
+    pub(crate) map: HashMap<String, Node<'a>>,
+}
```

Comment:
That's exactly right, in Rust you can only `impl` for types that are defined in the same crate. With a type alias I would be `impl`ing for `HashMap`, which wouldn't be possible since it is defined externally. `ComponentIndex` just didn't need any additional functionality other than the type safety it provides, so it's simpler to have it as an alias. For the `SheetIndex` I'm also using the `pub(crate)` visibility modifier for `map` to restrict external accessors accidentally modifying it.